### PR TITLE
fix(overlay plot): legend updates correctly when removing element via remove action

### LINKED ISSUE #7527: [Overlay Plots] Removing 1 Object from the Elements Pool Removes 2 from the Legend
#### Summary

In an Overlay Plot, removing an object from the elements pool is removing two objects from the legend. The two are being removed are the selected object and another random one. Sometimes, refreshing can bring the legend back to the correct state. 

Look at this image which was taken after removing an element. There are 7 elements in the legend, while there are 8 in the elements pool.
Screenshot 2024-02-27 at 2 42 18 PM

#### Expected vs Current Behavior

When you remove an object from the Elements Pool, we should see **only** that object being removed from the plot and the legend. 
#### Steps to Reproduce

1. Navigate to an Overlay Plot that is populated with multiple data points.
2. Count the number of elements in the plot. 
3. Go to Edit, and remove an object from the Elements Pool. 
4. Now go to the legend, and count the number of elements that are in it. 
5. Observe that there are 2 less elements in there, instead of 1. 

#### Environment

* Open MCT Version:   4.0.0-next
* Deployment Type:  testathon
* OS:
* Browser: Chrome

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### LINKED ISSUE #7530: [Overlay Plot] Deleting an element via Remove Action breaks the Legend
#### Summary

When removing an element in an Overlay Plot via the remove action and saving, errors are continually thrown when hovering over the plot (until refresh or nav away/back)

#### Expected vs Current Behavior

There should be no errors. The element should be removed from the plot and the legend should reflect that accurately. Highlights should still work.

#### Steps to Reproduce

1. Create an Overlay Plot with multiple SWGs in it
2. Save
3. Edit
4. Click on "Elements" tab to access the Elements Pool
5. Right click on an element, "Remove"
6. Save
7. Hover over the plot
8. Notice many errors in the console

#### Environment

* Open MCT Version:  4.0.0-next
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by ozyx**
Closes #7530 
Closes #7527 

### Describe your changes:

- Fixes an issue where removing a single item from an Overlay Plot via the Remove Action would remove 2 legend items
- Fixes an issue causing continuous errors to be thrown after removing an item from an Overlay Plot via the Remove action
- Small a11y improvement to legend items

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this a notable change that will require a special callout in the release notes [Notable Change](../docs/src/process/release.md) ? For example, will this break compatibility with existing APIs or projects which source these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: src/plugins/plot/legend/PlotLegend.vue [hunk 1]
```diff
@@ -165,7 +165,6 @@ export default {
       this.registerListeners(this.config);
     }
     this.listenTo(this.config.legend, 'change:expandByDefault', this.changeExpandDefault, this);
-    this.initialize();
```
Comment (ozyx) [inline src/plugins/plot/legend/PlotLegend.vue:168]: remove duplicate code

File: src/plugins/plot/legend/PlotLegendItemCollapsed.vue [hunk 2]
```diff
@@ -206,6 +206,9 @@ export default {
       const seriesIndexToRemove = this.seriesModels.findIndex(
         (series) => series.keyString === seriesToRemove.keyString
       );
+      if (seriesIndexToRemove === -1) {
+        return;
+      }
```
Comment (ozyx) [inline src/plugins/plot/legend/PlotLegendItemCollapsed.vue:211]: don't remove a seriesModel if we don't find one in the list

**Review by unlikelyzero — APPROVED** [review]
😎

File: src/plugins/plot/legend/PlotLegendItemCollapsed.vue (hunk 2 above)
Comment (jvigliotta) [reply src/plugins/plot/legend/PlotLegendItemCollapsed.vue:211]: haha, that ... _makes sense_

**Pull request merged**